### PR TITLE
Issue 2345: VersionedSerialization improvements 

### LINKED ISSUE #2345: VersionedSerializer improvements
Suggestions:
- `VersionedSerializer.SingleType.writeVersion()` should probably be renamed to `getWriteVersion()`
- `RevisionDataOutput`/`RevisionDataInput`
    - `writeArray`/`readArray` (supporting basic types; add a special one for `byte[]` since it has a direct implementation available already.

### PR DESCRIPTION
**Pull request by andreipaduroiu**
**Change log description**
Made a few improvements to VersionedSerialization
- Renamed `writeVersion()` to `getWriteVersion()` in `SingleType`
- Added support for encoding and decoding arrays.

**Purpose of the change**
Fixes #2345.

**What the code does**
New code for encoding/decoding arrays.
Renamed `writeVersion` to `getWriteVersion`

**How to verify it**
Unit tests.

### CODE REVIEW

File: common/src/test/java/io/pravega/common/io/serialization/RevisionDataStreamCommonTests.java [hunk 1]
```diff
@@ -313,6 +351,42 @@ private String getUTFString(int minChar, int maxChar) {
         return sb.toString();
     }
 
+    private <T> boolean arraysEqual(T[] a1, T[] a2) {
```
Comment (tkaitchuck) [inline common/src/test/java/io/pravega/common/io/serialization/RevisionDataStreamCommonTests.java]: Arrays.equals?

File: common/src/test/java/io/pravega/common/io/serialization/RevisionDataStreamCommonTests.java [hunk 2]
```diff
@@ -313,6 +351,42 @@ private String getUTFString(int minChar, int maxChar) {
...
+        for (int i = 0; i < a1.length; i++) {
+            if (!a1[i].equals(a2[i])) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private boolean byteArraysEqual(byte[] a1, byte[] a2) {
```
Comment (tkaitchuck) [inline common/src/test/java/io/pravega/common/io/serialization/RevisionDataStreamCommonTests.java]: Arrays.equals()

Comment (andreipaduroiu) [reply common/src/test/java/io/pravega/common/io/serialization/RevisionDataStreamCommonTests.java]: fixed

File: common/src/test/java/io/pravega/common/io/serialization/RevisionDataStreamCommonTests.java (hunk 1 above)
Comment (andreipaduroiu) [reply common/src/test/java/io/pravega/common/io/serialization/RevisionDataStreamCommonTests.java]: fixed

**Pull request merged**